weave/artifact_wandb.py

```diff
@@ -503,9 +516,16 @@ def add_artifact_reference(self, uri: str) -> None:
         # The URI for a get node might look like: wandb-artifact:///<entity>/<project>/test8:latest/obj
         # but the SDK add_reference call requires something like:
         # wandb-artifact://41727469666163743a353432353830303535/obj.object.json
-        assert uri.startswith("wandb-artifact:///")
-        uri_parts = WeaveWBArtifactURI.parse(uri)
-        uri_path = f"{uri_parts.entity_name}/{uri_parts.project_name}/{uri_parts.name}:{uri_parts.version}"
+        assert uri.startswith("wandb-artifact:///") or uri.startswith(
+            "wandb-artifact-by-id:///"
+        )
+        try:
+            uri_parts = WeaveWBArtifactURI.parse(uri)
```

Comment:
seems like rather than a try-catch here you could just use the scheme prefix to decide which path to take